### PR TITLE
Added documentation for MCP23017 component: binary sensor and switch …

### PR DESCRIPTION
**Pull request by jardiamj**
MCP23017

**Description:**
Documentation for new component MCP23017: binary sensor and switch platforms. In preparation for PR on home-assistant to be added as a new core component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#23127

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/mcp23017.markdown [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+sharing: true
+footer: true
+logo: raspberry-pi.png
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: 0.92
+ha_iot_class: Local Polling
+redirect_from:
```
Comment (klaasnicolaas) [inline source/_components/mcp23017.markdown]: Please remove, this is only for existing pages.

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_components/mcp23017.markdown (hunk 1 above)
Comment (jardiamj) [reply source/_components/mcp23017.markdown]: @klaasnicolaas, what do I need to remove? I'm sorry, I don't quite understand your statement.
Never mind, I see that you are referring to the redirect_from label. I apologize, my understanding of how the documentation is structured is still a bit limited.
EDIT: Label has been removed.

Comment (klaasnicolaas) [reply source/_components/mcp23017.markdown]: Yet you have solved it well 😉 

`redirect_from` is only needed if old / existing pages have to be referred to a new page, for example after merging single platform pages to 1 page and therefore old pages are removed and no longer exist.

File: source/_components/mcp23017.markdown [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+logo: raspberry-pi.png
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: 0.92
+ha_iot_class: Local Polling
+---
+
+The `mcp23017` component is the base for all related mcp23017 platforms in Home Assistant. There is no setup needed for the component itself, for the platforms please check their corresponding pages.
```
Comment (amelchio) [inline source/_components/mcp23017.markdown]:
```suggestion
The `mcp23017` component is the base for all related mcp23017 platforms in Home Assistant. There is no setup needed for the component itself, for the platforms please check their corresponding sections.
```

File: source/_components/mcp23017.markdown [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+## {% linkable_title Binary Sensor %}
+
+The `mcp23017` binary sensor platform allows you to read sensor values from the I/O pins of your [MCP23017 I2C I/O expander](https://www.adafruit.com/product/732).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port A (A1-A8) and 8-15 to port B (B1-B8).
+
+## {% linkable_title Configuration %}
+
+To use the I/O pins of an mcp23017 connected to and I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
```
Comment (amelchio) [inline source/_components/mcp23017.markdown]:
```suggestion
To use the I/O pins of an mcp23017 connected to an I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
```

File: source/_components/mcp23017.markdown [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+## {% linkable_title Switch %}
+
+The `mcp23017` switch platform allows you to write to the I/O pins of your [MCP23017 I2C I/O expander](https://www.adafruit.com/product/732).
+
+The pin numbers are from 0 to 15 where: 0-7 correspond to port A (A1-A8) and 8-15 to port B (B1-B8).
+
+## {% linkable_title Configuration %}
+
+To use the I/O pins of an mcp23017 connected to and I2C bus of your Raspberry Pi as switches, add the following to your `configuration.yaml` file:
```
Comment (amelchio) [inline source/_components/mcp23017.markdown]:
```suggestion
To use the I/O pins of an mcp23017 connected to an I2C bus of your Raspberry Pi as switches, add the following to your `configuration.yaml` file:
```

File: source/_components/mcp23017.markdown [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      required: true
+      type: [integer, string]
+invert_logic:
+  description: If true, inverts the output logic to ACTIVE LOW.
+  required: false
+  default: false
+  type: boolean
+{% endconfiguration %}
+
+For more details about the MCP23017 I2C I/O port expander you can find its datasheet here: [MCP23017](https://www.microchip.com/wwwproducts/en/MCP23017).
```
Comment (amelchio) [inline source/_components/mcp23017.markdown]: Maybe move this to the top so it is not repeated in each section.

**Review by amelchio — DISMISSED** [review]
Good, just a few typos.

Also, it looks odd to me that, for example, "Switch" and "Configuration" are separate sections at the same level but I am not well-versed in the documentation standards so it might be okay.

File: source/_components/mcp23017.markdown (hunk 5 above)
Comment (jardiamj) [reply source/_components/mcp23017.markdown]: I made the suggested changes. I also went through the documentation standards and there is nothing in regards to section levels.
I used an existing page as a template but it does make more sense to have "Configuration" as a sub-section of "Binary Sensor/Switch" so I did that as well.

**Review by klaasnicolaas — APPROVED** [review]

**Pull request merged**